### PR TITLE
AFHS-393 Updated postcode test to assert that invalid postcodes show …

### PR DESCRIPTION
**Pull request by dwybourn**
…up in their non sanitized format. Also added invalid postcodes with multiple spaces.

### CODE REVIEW

File: src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
 import org.openqa.selenium.WebElement;
+import uk.gov.dhsc.htbhf.page.NationalInsuranceNumberPage;
```
Comment (mikeSmith08) [inline src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java]: I'm puzzled why this import was added by your PR?

File: src/test/resources/features/acceptance/address-lookup.feature [hunk 2]
```diff
@@ -54,23 +54,27 @@ Feature: Select address
     Given I have entered my details up to the postcode page
     When I enter <postcode> as my postcode
     Then I am informed that the postcode is in the wrong format
+    Then I see the postcode: <postcode> in the input in the same format I entered it in
```
Comment (mikeSmith08) [inline src/test/resources/features/acceptance/address-lookup.feature]: This should be `And` not `Then`.

File: src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java (hunk 1 above)
Comment (dwybourn) [reply src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java]: Removed the unused import.

File: src/test/resources/features/acceptance/address-lookup.feature (hunk 2 above)
Comment (dwybourn) [reply src/test/resources/features/acceptance/address-lookup.feature]: Replaced with Then.

File: src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java [hunk 3]
```diff
@@ -128,6 +128,13 @@ public void assertPostcodeInWrongFormat() {
         assertPostcodeErrorPresent("Enter a correct postcode, like AA1 1AA");
     }
 
+    @Then("^I see the postcode: (.*) in the input in the same format I entered it in")
```
Comment (YetAnotherMatt) [inline src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java]: The wording on this isn't completely clear. How about `^The postcode remains in the same format as I entered it: (.*)$` or similar?

File: src/test/resources/features/acceptance/address-lookup.feature [hunk 4]
```diff
@@ -54,23 +54,27 @@ Feature: Select address
...
     When I enter <postcode> as my postcode
     Then I am informed that the postcode is in the wrong format
+    And I see the postcode: <postcode> in the input in the same format I entered it in
 
     Examples:
       | postcode |
       | AA1122BB |
       | A        |
       | 11AA21   |
+      | 11A  A21 |
```
Comment (YetAnotherMatt) [inline src/test/resources/features/acceptance/address-lookup.feature]: Can we have a postcode in lowercase too?

Comment (dwybourn) [reply src/test/resources/features/acceptance/address-lookup.feature]: Added postcode with lowercase text.

File: src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java (hunk 3 above)
Comment (dwybourn) [reply src/test/java/uk/gov/dhsc/htbhf/steps/AddressLookupSteps.java]: Replaced with 'I see the postcode: <postcode> in the same format I entered it in'

**Pull request merged**